.github/actions/linux-build/action.yml

```diff
@@ -71,6 +55,21 @@ runs:
             sudo apt-get -qq -y install clang-14
           fi
         fi
+    - name: ccache 
+      id: ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        # free total space for cache available on github is limited to 10GB, so only save on main branch;
+        # other branches will try to use cache from main, but branches that are too old won't benefit;
+        save: ${{ github.ref_name == 'main' }} 
+        # set enough space per build cache to keep a few previous versions to be used by older branches
+        # debug builds require more space
+        max-size: ${{ inputs.config == 'Release' && '200M' || '1G' }} 
```

Comment:
Where do these sizes come from? More importantly, is there some report or warning if or when these sizes become too small?

Reply:
basically, I looked at the size of the cache for one build and at least doubled it. debug builds require 300-400M for a single build. So there is room for 2-3 older versions. release builds are much smaller, <50M, so there is more room here. It's possible we want to tune those numbers later, give debug builds a bit more space. I have little data on how fast this will grow with the code changes we are making. 

It's important to stay below the 10G limit, otherwise caches will be evicted immediately after saving. Currently, there are 10 builds cached, so 1G on average per build is the absolute limit. 

I wanted to save a bit of room for caching other things, e.g. boost or other dependencies.

Reply:
That sounds reasonable. Maybe you could mention the 10G limit (or link the github page that mentions it) in the comment here.

The cache size is shown, but only under "Post Run" in a collapsed line "cache stats". I think it might be useful to make this more visible, there is a format that turns echo into an [annotation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-creating-an-annotation-for-an-error). An error if the cache is larger than expected would be nice.

Reply:
you can also see the cache sizes in the github web interface in Actions -> Cache.

The question is, how large is too large? The cache will never be bigger than the limit (e.g. 200M for release builds), because of automatic eviction. But after some code changes, the cache will certainly hit the limit and then never shrink again. The cache will remain around the limit. So only the cache hits and misses can give more information. But I don't know how many hits or misses are expected, because it depends on the changes in the code. One extreme case would be a less than 100% hit rate where there have been no changes since the last build. But that would be complicated to implement, comparing github log and time stamps of the cache, etc. Not convinced this is worth it. Maybe with more data there can be a heuristic.

Reply:
not sure what you mean. the size of the ccache directory will just be the limit set in the yaml (after some code changes of course)

Reply:
The idea is to try and look whether ccache tries to use more space than the action allows for. But if ccache limits its output/cache on its own, that wont work.